### PR TITLE
Update azure-storage-blob to 12.15.0: fix test flakiness

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### Description
Fix test flakiness
 
### Issues Resolved
Fixes https://github.com/opensearch-project/OpenSearch/issues/2782
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@tlfeng seems like I nailed it down, was running tests for the last hour continuously - 0 failures.

File: plugins/repository-azure/src/test/java/org/opensearch/repositories/azure/AzureBlobContainerRetriesTests.java [hunk 1]
```diff
@@ -282,7 +282,7 @@ public void testReadRangeBlobWithRetries() throws Exception {
                         exchange.getResponseHeaders().add("Content-Type", "application/octet-stream");
                         exchange.getResponseHeaders().add("Content-Length", String.valueOf(length));
                         exchange.getResponseHeaders()
-                            .add("Content-Range", String.format("bytes %d-%d/%d", rangeStart, rangeEnd.get(), bytes.length));
```
Comment (reta) [inline plugins/repository-azure/src/test/java/org/opensearch/repositories/azure/AzureBlobContainerRetriesTests.java:285]: Using format + `%d` was not a good idea - the locale is being taken into account

File: plugins/repository-azure/src/test/java/org/opensearch/repositories/azure/AzureBlobContainerRetriesTests.java [hunk 2]
```diff
@@ -232,7 +232,7 @@ public void testReadBlobWithRetries() throws Exception {
                         exchange.getResponseHeaders().add("Content-Length", String.valueOf(length));
                         exchange.getResponseHeaders().add("x-ms-blob-type", "blockblob");
                         exchange.getResponseHeaders()
-                            .add("Content-Range", String.format("bytes %d-%d/%d", rangeStart, bytes.length, bytes.length));
+                            .add("Content-Range", String.format("bytes " + rangeStart + "-" + bytes.length + "/" + bytes.length));
```
Comment (andrross) [inline plugins/repository-azure/src/test/java/org/opensearch/repositories/azure/AzureBlobContainerRetriesTests.java:235]: Is the `String.format()` doing anything now?

Comment (reta) [reply plugins/repository-azure/src/test/java/org/opensearch/repositories/azure/AzureBlobContainerRetriesTests.java:235]: Oh ... definitely not, I forgot to remove it, thank!

**Review by kotwanikunal — APPROVED** [review]

**Review by andrross — APPROVED** [review]
Just curious, how could the locale change the formatting of an `int` using `%d`? It makes me question if String.format() should really ever be used without specifying a locale.

**Comment by tlfeng** [comment]
Looks like the "forbidden API checker" is not applied to this sub-project.
For most other directories, if the locale is not assigned to String.format(), it won't pass the check.
For example:
```
> Task :client:rest:forbiddenApisTest FAILED
Forbidden method invocation: java.lang.String#format(java.lang.String,java.lang.Object[]) [Uses default locale]
  in org.opensearch.client.ResponseExceptionTests (ResponseExceptionTests.java:93)
Scanned 66 class file(s) for forbidden API invocations (in 0.30s), 1 error(s).
```
But it's interesting that the `int` variable can become a character other than number. 😄

**Comment by reta** [comment]
> Just curious, how could the locale change the formatting of an `int` using `%d`? It makes me question if String.format() should really ever be used without specifying a locale.

As per javadoc [1] , `String#format`:

> The locale always used is the one returned by [Locale.getDefault()](https://docs.oracle.com/javase/8/docs/api/java/util/Locale.html#getDefault--).

[1] https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#format-java.lang.String-java.lang.Object...-

**Comment by reta** [comment]
> Looks like the "forbidden API checker" is not applied to this sub-project.

That could be, since it is only in test classes.

> But it's interesting that the `int` variable can become a character other than number. 😄

Yes, it actually does!

**Pull request merged**